### ISSUE TITLE
Crash on terraform plan that used to work

### ISSUE THREAD

**Issue by rsangiorgi1** [terraform-providers/terraform-provider-azurerm#5163]
I get a terraform crash while running terraform plan or refresh (init works though). In the log it seems to be related to my logic app (called workflow1) so I added the tf module for my logic app and alerts here.

Please let me know what I should do to go back to (any) working way. I tried commenting out all the resources that I thought might cause trouble, but even then I get a crash when running terraform plan. Should I do something to my tf state? Also, please let me know if you need more info, I'm new to the community. Thanks!

### Terraform (and AzureRM Provider) Version

Version info:
```
Terraform v0.12.17
+ provider.azurerm v1.38.0
+ provider.local v1.4.0
```

### Terraform Configuration Files

```hcl
variable rg_name {}
variable location {}
variable vm_name {}
variable vm_id {}

data "local_file" "foo" {
    filename = "schemas/schema.json"
}

data "local_file" "custom_action_body" {
    filename = "schemas/slackAction.json"
}

data "local_file" "message_to_slack" {
    filename = "schemas/messageToSlack.json"
}

resource "azurerm_monitor_action_group" "slack_ag" {
  name                = "slack-ag"
  resource_group_name = var.rg_name
  short_name          = "exampleact"

  # all alerts connecting this action group will trigger the following logic apps
  logic_app_receiver {
    name                    = azurerm_logic_app_workflow.example.name
    resource_id             = azurerm_logic_app_workflow.example.id
    callback_url            = azurerm_logic_app_workflow.example.access_endpoint
    use_common_alert_schema = true
  }
}

# logic app
resource "azurerm_logic_app_workflow" "example" {
  name                = "workflow1"
  location            = var.location
  resource_group_name = var.rg_name
}

resource "azurerm_logic_app_trigger_http_request" "example" {
  name         = "some-http-trigger"
  logic_app_id = azurerm_logic_app_workflow.example.id

  schema = data.local_file.foo.content
}

resource "azurerm_logic_app_action_http" "example" {
  name         = "webhook"
  logic_app_id = azurerm_logic_app_workflow.example.id
  method       = "POST"
  body         = data.local_file.message_to_slack.content
  uri          = "https://hooks.slack.com/services/myslackwebhook....."
}

# Alerts
resource "azurerm_monitor_metric_alert" "example" {
  name                = "${var.vm_name}-cpu"
  resource_group_name = var.rg_name
  # a list of which VMs to watch
  scopes              = [var.vm_id]
  description         = "An alert rule to watch the metric Percentage CPU"

  criteria {
    metric_namespace = "Microsoft.Compute/virtualMachines"
    metric_name = "Percentage CPU"
    operator    = "GreaterThan"
    threshold   = 1
    aggregation = "Average"
  }

  action {
    action_group_id = azurerm_monitor_action_group.slack_ag.id
  }
}

output "logic_app_url" {
  value       = azurerm_logic_app_workflow.example.access_endpoint
  description = "endpoint to trigger logic app"
}
```

### Debug Output

```
terraform refresh -var-file="dev.secrets.tfvars" -var-file="dev.tfvars"
module.monitoring.data.local_file.custom_action_body: Refreshing state...
module.monitoring.data.local_file.message_to_slack: Refreshing state...
module.monitoring.data.local_file.foo: Refreshing state...
azurerm_resource_group.kb_ozp_env_rg: Refreshing state... [id=/subscriptions/x/resourceGroups/kb-ozp-dev-rg]
azurerm_public_ip.kb_ozp_public_ip: Refreshing state... [id=/subscriptions/x/resourceGroups/kb-ozp-dev-rg/providers/Microsoft.Network/publicIPAddresses/kb-ozp-tf-public-ip]
module.monitoring.azurerm_logic_app_workflow.example: Refreshing state... [id=/subscriptions/x/resourceGroups/kb-ozp-dev-rg/providers/Microsoft.Logic/workflows/workflow1]
azurerm_virtual_network.kb_ozp_vnet: Refreshing state... [id=/subscriptions/subscriptions/x/resourceGroups/kb-ozp-dev-rg/providers/Microsoft.Network/virtualNetworks/kb-ozp-tf-vnet]
azurerm_network_security_group.kb_ozp_network_security_group: Refreshing state... [id=/subscriptions/3/resourceGroups/kb-ozp-dev-rg/providers/Microsoft.Network/networkSecurityGroups/kb-ozp-tf-network-security-group]
azurerm_subnet.kb_ozp_subnet: Refreshing state... [id=/subscriptions/xresourceGroups/kb-ozp-dev-rg/providers/Microsoft.Network/virtualNetworks/kb-ozp-tf-vnet/subnets/kb-ozp-tf-subnet]
module.monitoring.azurerm_logic_app_action_http.example: Refreshing state... [id=/subscriptions/e193ex3/resourceGroups/kb-ozp-dev-rg/providers/Microsoft.Logic/workflows/workflow1/actions/webhook]
module.monitoring.azurerm_monitor_action_group.slack_ag: Refreshing state... [id=/subscriptions/ex23/resourceGroups/kb-ozp-dev-rg/providers/microsoft.insights/actionGroups/slack-ag]
module.monitoring.azurerm_logic_app_trigger_http_request.example: Refreshing state... [id=/subscriptions/ex23/resourceGroups/kb-ozp-dev-rg/providers/Microsoft.Logic/workflows/workflow1/triggers/some-http-trigger]

Error: rpc error: code = Unavailable desc = transport is closing



Error: rpc error: code = Unavailable desc = transport is closing



Error: rpc error: code = Unavailable desc = transport is closing



Error: rpc error: code = Unavailable desc = transport is closing


panic: interface conversion: interface {} is map[string]interface {}, not string
2019-12-13T18:05:18.858+0100 [DEBUG] plugin.terraform-provider-azurerm_v1.38.0_x4: 
2019-12-13T18:05:18.858+0100 [DEBUG] plugin.terraform-provider-azurerm_v1.38.0_x4: goroutine 96 [running]:
2019-12-13T18:05:18.858+0100 [DEBUG] plugin.terraform-provider-azurerm_v1.38.0_x4: github.com/terraform-providers/terraform-provider-azurerm/azurerm.resourceArmLogicAppActionHTTPRead(0xc0001f2000, 0x35b7b80, 0xc000608400, 0xc0001f2000, 0x0)
2019-12-13T18:05:18.858+0100 [DEBUG] plugin.terraform-provider-azurerm_v1.38.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/resource_arm_logic_app_action_http.go:162 +0xefe
2019-12-13T18:05:18.858+0100 [DEBUG] plugin.terraform-provider-azurerm_v1.38.0_x4: github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc00062a180, 0xc00095f860, 0x35b7b80, 0xc000608400, 0xc000e16bc0, 0x0, 0x0)
2019-12-13T18:05:18.858+0100 [DEBUG] plugin.terraform-provider-azurerm_v1.38.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/resource.go:455 +0x119
2019-12-13T18:05:18.858+0100 [DEBUG] plugin.terraform-provider-azurerm_v1.38.0_x4: github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ReadResource(0xc000152f08, 0x4405340, 0xc000809b60, 0xc00095f770, 0xc000152f08, 0xc000809b60, 0xc000e40bd0)
2019-12-13T18:05:18.858+0100 [DEBUG] plugin.terraform-provider-azurerm_v1.38.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin/grpc_provider.go:525 +0x3d7
2019-12-13T18:05:18.858+0100 [DEBUG] plugin.terraform-provider-azurerm_v1.38.0_x4: github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ReadResource_Handler(0x3a9eaa0, 0xc000152f08, 0x4405340, 0xc000809b60, 0xc0001603c0, 0x0, 0x4405340, 0xc000809b60, 0xc000616500, 0x265)
2019-12-13T18:05:18.858+0100 [DEBUG] plugin.terraform-provider-azurerm_v1.38.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5/tfplugin5.pb.go:3153 +0x23e
2019-12-13T18:05:18.858+0100 [DEBUG] plugin.terraform-provider-azurerm_v1.38.0_x4: github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc.(*Server).processUnaryRPC(0xc0003c4420, 0x4458b40, 0xc000587e00, 0xc000862100, 0xc000808480, 0x7e915f0, 0x0, 0x0, 0x0)
2019-12-13T18:05:18.858+0100 [DEBUG] plugin.terraform-provider-azurerm_v1.38.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:995 +0x466
2019-12-13T18:05:18.858+0100 [DEBUG] plugin.terraform-provider-azurerm_v1.38.0_x4: github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc.(*Server).handleStream(0xc0003c4420, 0x4458b40, 0xc000587e00, 0xc000862100, 0x0)
2019-12-13T18:05:18.858+0100 [DEBUG] plugin.terraform-provider-azurerm_v1.38.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1275 +0xda6
2019-12-13T18:05:18.858+0100 [DEBUG] plugin.terraform-provider-azurerm_v1.38.0_x4: github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc00075c100, 0xc0003c4420, 0x4458b40, 0xc000587e00, 0xc000862100)
2019-12-13T18:05:18.858+0100 [DEBUG] plugin.terraform-provider-azurerm_v1.38.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:710 +0x9f
2019-12-13T18:05:18.858+0100 [DEBUG] plugin.terraform-provider-azurerm_v1.38.0_x4: created by github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc.(*Server).serveStreams.func1
2019-12-13T18:05:18.858+0100 [DEBUG] plugin.terraform-provider-azurerm_v1.38.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:708 +0xa1
2019-12-13T18:05:18.865+0100 [DEBUG] plugin: plugin process exited: path=/home/renee/dev/repos/kb/ozp-infra/.terraform/plugins/linux_amd64/terraform-provider-azurerm_v1.38.0_x4 pid=28845 error="exit status 2"
2019/12/13 18:05:18 [ERROR] module.monitoring: eval: *terraform.EvalRefresh, err: rpc error: code = Unavailable desc = transport is closing
2019/12/13 18:05:18 [ERROR] module.monitoring: eval: *terraform.EvalSequence, err: rpc error: code = Unavailable desc = transport is closing
2019/12/13 18:05:18 [ERROR] module.monitoring: eval: *terraform.EvalRefresh, err: rpc error: code = Unavailable desc = transport is closing
2019/12/13 18:05:18 [TRACE] [walkRefresh] Exiting eval tree: module.monitoring.azurerm_logic_app_action_http.example
2019/12/13 18:05:18 [ERROR] module.monitoring: eval: *terraform.EvalSequence, err: rpc error: code = Unavailable desc = transport is closing
2019/12/13 18:05:18 [TRACE] [walkRefresh] Exiting eval tree: module.monitoring.azurerm_logic_app_trigger_http_request.example
2019/12/13 18:05:18 [ERROR] module.monitoring: eval: *terraform.EvalRefresh, err: rpc error: code = Unavailable desc = transport is closing
2019/12/13 18:05:18 [TRACE] vertex "module.monitoring.azurerm_logic_app_trigger_http_request.example": visit complete
2019/12/13 18:05:18 [TRACE] vertex "module.monitoring.azurerm_logic_app_trigger_http_request.example": dynamic subgraph encountered errors
2019/12/13 18:05:18 [TRACE] vertex "module.monitoring.azurerm_logic_app_trigger_http_request.example": visit complete
2019/12/13 18:05:18 [TRACE] vertex "module.monitoring.azurerm_logic_app_action_http.example": visit complete
2019/12/13 18:05:18 [TRACE] vertex "module.monitoring.azurerm_logic_app_action_http.example": dynamic subgraph encountered errors
2019/12/13 18:05:18 [TRACE] vertex "module.monitoring.azurerm_logic_app_action_http.example": visit complete
2019/12/13 18:05:18 [ERROR] module.monitoring: eval: *terraform.EvalSequence, err: rpc error: code = Unavailable desc = transport is closing
2019/12/13 18:05:18 [TRACE] [walkRefresh] Exiting eval tree: module.monitoring.azurerm_monitor_action_group.slack_ag
2019/12/13 18:05:18 [TRACE] vertex "module.monitoring.azurerm_monitor_action_group.slack_ag": visit complete
2019/12/13 18:05:18 [TRACE] vertex "module.monitoring.azurerm_monitor_action_group.slack_ag": dynamic subgraph encountered errors
2019/12/13 18:05:18 [TRACE] vertex "module.monitoring.azurerm_monitor_action_group.slack_ag": visit complete
2019/12/13 18:05:18 [ERROR] <root>: eval: *terraform.EvalRefresh, err: rpc error: code = Unavailable desc = transport is closing
2019/12/13 18:05:18 [ERROR] <root>: eval: *terraform.EvalSequence, err: rpc error: code = Unavailable desc = transport is closing
2019/12/13 18:05:18 [TRACE] [walkRefresh] Exiting eval tree: azurerm_subnet.kb_ozp_subnet
2019/12/13 18:05:18 [TRACE] vertex "azurerm_subnet.kb_ozp_subnet": visit complete
2019/12/13 18:05:18 [TRACE] vertex "azurerm_subnet.kb_ozp_subnet": dynamic subgraph encountered errors
2019/12/13 18:05:18 [TRACE] vertex "azurerm_subnet.kb_ozp_subnet": visit complete
2019/12/13 18:05:18 [TRACE] dag/walk: upstream of "azurerm_network_interface.kb_ozp_network_interface" errored, so skipping
2019/12/13 18:05:18 [TRACE] dag/walk: upstream of "azurerm_virtual_machine.kb_ozp_env_virtual_machine" errored, so skipping
2019/12/13 18:05:18 [TRACE] dag/walk: upstream of "module.monitoring.var.vm_id" errored, so skipping
2019/12/13 18:05:18 [TRACE] dag/walk: upstream of "module.monitoring.var.vm_name" errored, so skipping
2019/12/13 18:05:18 [TRACE] dag/walk: upstream of "module.monitoring.azurerm_monitor_metric_alert.example" errored, so skipping
2019/12/13 18:05:18 [TRACE] dag/walk: upstream of "provider.azurerm (close)" errored, so skipping
2019/12/13 18:05:18 [TRACE] dag/walk: upstream of "root" errored, so skipping
2019/12/13 18:05:18 [INFO] backend/local: refresh calling Refresh
2019-12-13T18:05:18.951+0100 [DEBUG] plugin: plugin exited
```

### Panic Output

```
2019-12-13T18:05:18.855+0100 [DEBUG] plugin.terraform-provider-azurerm_v1.38.0_x4: [DEBUG] Unlocking "azurerm_logic_app.workflow1"
2019-12-13T18:05:18.855+0100 [DEBUG] plugin.terraform-provider-azurerm_v1.38.0_x4: [DEBUG] Unlocked "azurerm_logic_app.workflow1"
2019-12-13T18:05:18.855+0100 [DEBUG] plugin.terraform-provider-azurerm_v1.38.0_x4: [DEBUG] Locked "azurerm_logic_app.workflow1"
2019-12-13T18:05:18.858+0100 [DEBUG] plugin.terraform-provider-azurerm_v1.38.0_x4: panic: interface conversion: interface {} is map[string]interface {}, not string
```

### Steps to Reproduce

1. `terraform apply`

### Important Factoids
My tf state is in Azure, I am running tf locally using the Azure CLI.

### References

* #0000

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by AlexandreSi** [comment]
Hi there,
I seem to have a similar problem here, though not on the same resource.

### Terraform (and AzureRM Provider) Version

```
Terraform v0.12.18
+ provider.azuread v0.7.0
+ provider.azurerm v1.39.0
```

### Affected Resource(s)



* `azurerm_app_service`

### Terraform Configuration Files



```hcl
resource "azurerm_container_registry" "acr" {
  name                     = "${terraform.workspace}ContainerRegistry"
  resource_group_name      = azurerm_resource_group.project.name
  location                 = azurerm_resource_group.project.location
  sku                      = "Basic"
  admin_enabled            = true
}

resource "azurerm_app_service_plan" "api" {
  name                     = "api"
  resource_group_name      = azurerm_resource_group.project.name
  location                 = azurerm_resource_group.project.location
  kind                     = "Linux"

  # Mandatory for Linux plans
  reserved                 = true
  sku {
    tier                   = "PremiumV2"
    size                   = "P1v2"
  }
}

resource "azurerm_app_service" "api" {
  name                     = "api-${terraform.workspace}"
  resource_group_name      = azurerm_resource_group.project.name
  location                 = azurerm_resource_group.project.location
  app_service_plan_id      = azurerm_app_service_plan.api.id

  site_config {
    always_on              = true
    linux_fx_version       = "DOCKER|${azurerm_container_registry.acr.login_server}/project-api:latest"
  }

  app_settings = {
    // settings for app service
    WEBSITES_ENABLE_APP_SERVICE_STORAGE = false
    WEBSITES_PORT = 5000
    SSH_PORT = 2222
    DOCKER_REGISTRY_SERVER_URL = "https://${azurerm_container_registry.acr.login_server}"
    DOCKER_REGISTRY_SERVER_USERNAME = azurerm_container_registry.acr.admin_username
    DOCKER_REGISTRY_SERVER_PASSWORD = azurerm_container_registry.acr.admin_password
    // env variables for the application
    DATABASE_NAME = var.database_name
    DATABASE_USER = "${var.database_user}@${azurerm_postgresql_server.db.name}"
    DATABASE_PASSWORD = var.database_password
    DATABASE_HOST = azurerm_postgresql_server.db.fqdn
    DATABASE_PORT = 5432
    APP_SECRET_KEY = var.api_secret_key[terraform.workspace]
    APP_ALLOWED_HOSTS = "*"
    LOG_LEVEL = "INFO"
    USE_S3 = true
    AWS_ACCESS_KEY_ID = var.aws_access_key_id_django
    AWS_SECRET_ACCESS_KEY = var.aws_secret_access_key_django
    AWS_STORAGE_BUCKET_NAME = "${var.static_storage_bucket_name_prefix}-${terraform.workspace}"
    BASE_URL = "https://${var.api-base-url[terraform.workspace]}"
  }
}
```

### Debug Output

```
data.azurerm_storage_account.client: Refreshing state...
azurerm_app_service.api: Creating...
azurerm_app_service.api: Still creating... [10s elapsed]
azurerm_app_service.api: Still creating... [20s elapsed]

Error: rpc error: code = Unavailable desc = transport is closing

[...]
2020/01/06 14:40:44 [DEBUG] azurerm_app_service.api: apply errored, but we're indicating that via the Error pointer rather than returning it: rpc error: code = Unavailable desc = transport is closing
2020/01/06 14:40:44 [TRACE] <root>: eval: *terraform.EvalMaybeTainted
2020/01/06 14:40:44 [TRACE] EvalMaybeTainted: azurerm_app_service.api encountered an error during creation, so it is now marked as tainted
2020/01/06 14:40:44 [TRACE] <root>: eval: *terraform.EvalWriteState
2020/01/06 14:40:44 [TRACE] EvalWriteState: removing state object for azurerm_app_service.api
2020/01/06 14:40:44 [TRACE] <root>: eval: *terraform.EvalApplyProvisioners
2020/01/06 14:40:44 [TRACE] EvalApplyProvisioners: azurerm_app_service.api has no state, so skipping provisioners
2020/01/06 14:40:44 [TRACE] <root>: eval: *terraform.EvalMaybeTainted
2020/01/06 14:40:44 [TRACE] EvalMaybeTainted: azurerm_app_service.api encountered an error during creation, so it is now marked as tainted
2020/01/06 14:40:44 [TRACE] <root>: eval: *terraform.EvalWriteState
2020/01/06 14:40:44 [TRACE] EvalWriteState: removing state object for azurerm_app_service.api
2020/01/06 14:40:44 [TRACE] <root>: eval: *terraform.EvalIf
2020/01/06 14:40:44 [TRACE] <root>: eval: *terraform.EvalIf
2020/01/06 14:40:44 [TRACE] <root>: eval: *terraform.EvalWriteDiff
2020/01/06 14:40:44 [TRACE] <root>: eval: *terraform.EvalApplyPost
2020/01/06 14:40:44 [ERROR] <root>: eval: *terraform.EvalApplyPost, err: rpc error: code = Unavailable desc = transport is closing
2020/01/06 14:40:44 [ERROR] <root>: eval: *terraform.EvalSequence, err: rpc error: code = Unavailable desc = transport is closing
2020/01/06 14:40:44 [TRACE] [walkApply] Exiting eval tree: azurerm_app_service.api
2020/01/06 14:40:44 [TRACE] vertex "azurerm_app_service.api": visit complete
2020/01/06 14:40:44 [TRACE] dag/walk: upstream of "provider.azurerm (close)" errored, so skipping
2020/01/06 14:40:44 [TRACE] dag/walk: upstream of "meta.count-boundary (EachMode fixup)" errored, so skipping
2020/01/06 14:40:44 [TRACE] dag/walk: upstream of "root" errored, so skipping
2020-01-06T14:40:44.216+0100 [DEBUG] plugin: plugin exited
```

### Panic Output

```
panic: interface conversion: interface {} is nil, not map[string]interface {}
```

### Expected Behavior

Terraform apply exits normally.

### Actual Behavior

Crash

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

We tried to manually create the resource and then import it. After removing a few attributes, we tried to apply the configuration but it failed the same way.

**Comment by tombuildsstuff** [comment]
@AlexandreSi mind opening a separate issue?

**Comment by ms-henglu** [comment]
Here're the steps to reproduce the crash, I think it can be reproduced in any version.

1. Deploy azurerm_logic_app_workflow and azurerm_logic_app_action_http, you can use the following example.

provider "azurerm" {
  features {}
}
resource "azurerm_resource_group" "rg" {
  name     = "rg-henglu-app"
  location = "westus2"
}
resource "azurerm_logic_app_workflow" "workflow" {
  name                = "ameren-MisUpdateWorkflow-henglgu"
  location            = azurerm_resource_group.rg.location
  resource_group_name = azurerm_resource_group.rg.name
}
resource "azurerm_logic_app_action_http" "misupdateaction" {
  name         = "ameren-misupdate-webhook-henglgu"
  logic_app_id = azurerm_logic_app_workflow.workflow.id
  method       = "POST"
  uri          = "api_url/api/Mis/Update"
  body         = <<BODY
{
  "secret" : "123"
}
BODY
}

 

2. Open Azure Portal, and check the azurerm_logic_app_workflow, Click edit button

3. Click Code View, then click Save button. 

4. Run any terraform command, ex: terraform refresh, it will crash.

 

Cause:
In step3, although user may not change any codes in code view, but the attribute body's format is changed, then it caused crash.

Solution:

1. Remove the body which is not compatible.
1.1 Open Code View like above steps
1.2 Delete property body and its value

2. Now running terraform commands won't cause crash. You can remove body from Terraform codes and apply it, then add it back and apply it again.

**Comment by tombuildsstuff** [comment]
Fixed via #11847
